### PR TITLE
[SPARK-44535][CONNECT][SQL] Move required Streaming API to sql/api

### PR DESCRIPTION
**Pull request by hvanhovell**
### What changes were proposed in this pull request?
This PR moves a bunch streaming classed to the SQL/API project.

### Why are the changes needed?
This is needed to disconnect the Spark Connect Scala Client from catalyst.

### Does this PR introduce _any_ user-facing change?
No.

### How was this patch tested?
Existing Tests.

### CODE REVIEW

**Comment by hvanhovell** [comment]
@rangadi PTAL

**Review by zhenlineo — APPROVED** [review]
LGTM

File: connector/connect/client/jvm/src/main/scala/org/apache/spark/sql/streaming/StreamingQueryListener.scala [hunk 1]
```diff
@@ -116,7 +115,7 @@ object StreamingQueryListener extends Serializable {
    * @since 3.5.0
    */
   @Evolving
-  trait Event extends SparkListenerEvent
+  trait Event
```
Comment (rangadi) [inline connector/connect/client/jvm/src/main/scala/org/apache/spark/sql/streaming/StreamingQueryListener.scala:118]: @bogao007 PTAL this change.

**Review by rangadi — APPROVED** [review]
LGTM. Looks safe to me, assuming the tests pass.

**Comment by hvanhovell** [comment]
Merging this. Python tests are known to be flaky.

**Pull request closed without merging**